dipy/tracking/streamlinespeed.pyx

```diff
@@ -102,15 +102,18 @@ def length(streamlines):
         arclengths = np.zeros(len(streamlines), dtype=np.float64)
 
         if streamlines.data.dtype == np.float32:
-            c_arclengths_from_arraysequence[float2d](streamlines.data,
-                                                     streamlines._offsets,
```

Comment:
What type are the offsets, lengths normally?  I guess they should be `np.intp` type.  I guess this conversion could have a cost, most efficient might be guarantee their type on streamline object creation, then you can do `streamlines._offsets.astype(np.np_intp)` here assuming it will have a very small cost (just the type check).